generators/init/index.js

```diff
@@ -0,0 +1,258 @@
+        await this.prompt(
+          [
+            {
+              name: 'projectName',
+              when: () => !this.abort,
+              type: 'input',
+              message: 'What is the project name of your application?',
+              default: () => this._getDefaultProjectName(),
+            },
+            {
+              name: 'baseName',
+              when: () => !this.abort,
+              type: 'input',
+              validate: input => this._validateBaseName(input),
+              message: 'What is the base name of your application?',
+              default: () => this.getDefaultAppName(),
+            },
+            {
+              name: 'prettierDefaultIndent',
+              when: () => !this.abort,
+              type: 'input',
+              message: 'What is the default indentation?',
+              default: defaultConfig.prettierDefaultIndent,
+            },
+            {
+              name: 'prettierJavaIndent',
+              when: () => !this.abort,
+              type: 'input',
+              message: 'What is the Java indentation?',
+              default: defaultConfig.prettierJavaIndent,
+            },
+          ],
+          this.config
+        );
+      },
+    };
+  }
+
+  get prompting() {
+    if (this.fromBlueprint) return;
+    return this._prompting();
+  }
+
+  _configuring() {
+    return {
+      setDefaults() {
+        this.config.defaults({
+          baseName: this.getDefaultAppName(),
+          ...requiredConfig,
+        });
+      },
+    };
+  }
+
+  get configuring() {
+    if (this.fromBlueprint) return;
+    return this._configuring();
+  }
+
+  _loading() {
+    return {
+      loadConfig() {
+        this.loadInitConfig();
+      },
+      loadDerivedConfig() {
+        this.loadDerivedInitConfig();
+      },
+      loadConstant() {
+        this.NODE_VERSION = constants.NODE_VERSION;
+      },
+      loadDependabotDependencies() {
+        this.loadDependabotDependencies(this.fetchFromInstalledJHipster('init', 'templates', 'package.json'));
+      },
+    };
+  }
+
+  get loading() {
+    if (this.fromBlueprint) return;
+    return this._loading();
+  }
+
+  _writing() {
+    return {
+      writeFiles() {
+        return this.writeFilesToDisk(files);
+      },
+    };
+  }
+
+  get writing() {
+    if (this.fromBlueprint) return;
+    return this._writing();
+  }
+
+  _install() {
+    return {
+      // Initialize git repository before package manager install for commit hooks
+      async initGitRepo() {
+        if (this.options.skipGit) return;
+        if (!this.isGitInstalled()) {
+          this.warning('Git repository could not be initialized, as Git is not installed on your system');
+          return;
+        }
+        try {
+          const git = simpleGit({ baseDir: this.destinationPath() });
+          this.gitInitialized = (await git.checkIsRepo()) || ((await git.init()) && true);
+          this.log(chalk.green.bold('Git repository initialized.'));
```

Comment:
this part doesn't work on my computer, I don't understand why, as the CI passed.
I changed to:

```js
          git.checkIsRepo().then(
            () => this.log(chalk.green.bold('Git repository already initialized.')),
            () => {
              git.init().then(() => this.log(chalk.green.bold('Git repository initialized.')));
            }
          );
          this.gitInitialized = true;
```

What do you think ?

Reply:
git.checkIsRepo() returns false.
It shouldn’t throw.
Can you post the stack trace?

Reply:
Here my full log:

```
➜ npm run test test/init.spec.js

> generator-jhipster@7.1.0 pretest
> npm run lint


> generator-jhipster@7.1.0 lint
> npm run eslint && npm run ejs-lint


> generator-jhipster@7.1.0 eslint
> eslint .


> generator-jhipster@7.1.0 ejs-lint
> mocha test/ejslint.js



  EJS Lint
    ✔ EJS templates are valid (597ms)


  1 passing (603ms)


> generator-jhipster@7.1.0 test
> mocha --no-insight --forbid-only --parallel "test/init.spec.js"



  JHipster init generator
    with default options
      ✔ should create expected files (3ms)
    with custom prompt values
      and default options
        ✔ should write custom config to .yo-rc.json
      and skipPrompts option
        ✔ should write default values to .yo-rc.json (1ms)
      and existing config
        ✔ should not write custom prompt values
      and askAnswered option on an exiting project
        ✔ should write custom prompt values
    with custom
      skipCommitHook option
        ✔ should create expected files (1ms)
    git feature
      with default option
        1) should create .git
        2) should create 1 commit
      with skipGit option
        ✔ should not create .git (1ms)
      regenerating
        3) should have 1 commit
        4) should have uncommited files


  7 passing (3s)
  4 failing

  1) JHipster init generator
       git feature
         with default option
           should create .git:
     Error: expect(received).resolves.toBeUndefined()

Received promise rejected instead of resolved
Rejected to value: [Error: ENOENT: no such file or directory, access '/tmp/506360400a726250d7f56dc1c15843285b72df73/.git']
      at expect (node_modules/expect/build/index.js:135:15)
      at Context.<anonymous> (test/init.spec.js:94:15)
      at processImmediate (internal/timers.js:464:21)

  2) JHipster init generator
       git feature
         with default option
           should create 1 commit:
     Error: expect(received).resolves.toMatchObject()

Received promise rejected instead of resolved
Rejected to value: [Error: fatal: ni ceci ni aucun de ses répertoires parents n'est un dépôt git : .git
]
      at expect (node_modules/expect/build/index.js:135:15)
      at Context.<anonymous> (test/init.spec.js:98:15)
      at processImmediate (internal/timers.js:464:21)

  3) JHipster init generator
       git feature
         regenerating
           should have 1 commit:
     Error: expect(received).resolves.toMatchObject()

Received promise rejected instead of resolved
Rejected to value: [Error: fatal: ni ceci ni aucun de ses répertoires parents n'est un dépôt git : .git
]
      at expect (node_modules/expect/build/index.js:135:15)
      at Context.<anonymous> (test/init.spec.js:121:15)
      at processImmediate (internal/timers.js:464:21)

  4) JHipster init generator
       git feature
         regenerating
           should have uncommited files:
     Error: expect(received).resolves.toMatch()

Received promise rejected instead of resolved
Rejected to value: [Error: warning: Pas un dépôt git. Utilisez --no-index pour comparer deux chemins hors d'un arbre de travail
usage : git diff --no-index [<options>] <chemin> <chemin>·
Options de format de sortie de diff
    -p, --patch           générer la rustine
    -s, --no-patch        supprimer la sortie des différences
    -u                    générer la rustine
    -U, --unified[=<n>]   générer les diffs avec <n> lignes de contexte
    -W, --function-context
                          générer les diffs avec <n> lignes de contexte
    --raw                 générer le diff en format brut
    --patch-with-raw      synonyme de '-p --raw'
    --patch-with-stat     synonyme de '-p --stat'
    --numstat             --stat pour traitement automatique
    --shortstat           afficher seulement la dernière ligne de --stat
    -X, --dirstat[=<param1,param2>...]
                          afficher la distribution des quantités de modifications relatives pour chaque sous-répertoire
    --cumulative          synonyme pour --dirstat=cumulative
    --dirstat-by-file[=<param1,param2>...]
                          synonyme pour --dirstat=files,param1,param2...
    --check               avertir si les modifications introduisent des marqueurs de conflit ou des erreurs d'espace
    --summary             résumé succinct tel que les créations, les renommages et les modifications de mode
    --name-only           n'afficher que les noms de fichiers modifiés
    --name-status         n'afficher que les noms et les status des fichiers modifiés
    --stat[=<largeur>[,<largeur-de-nom>[,<compte>]]]
                          générer un diffstat
    --stat-width <largeur>
                          générer un diffstat avec la largeur indiquée
    --stat-name-width <largeur>
                          génerer un diffstat avec la largeur de nom indiquée
    --stat-graph-width <largeur>
                          génerer un diffstat avec la largeur de graphe indiquée
    --stat-count <compte>
                          générer un diffstat avec des lignes limitées
    --compact-summary     générer une résumé compact dans le diffstat
    --binary              produire un diff binaire qui peut être appliqué
    --full-index          afficher les noms complets des objets pre- et post-image sur les lignes "index"
    --color[=<quand>]     afficher un diff coloré
    --ws-error-highlight <sorte>
                          surligner les erreurs d'espace dans les lignes 'contexte', 'ancien', 'nouveau' dans le diff
    -z                    ne pas compresser les chemins et utiliser des NULs comme terminateurs de champs dans --raw ou --numstat
    --abbrev[=<n>]        utiliser <n> chiffres pour afficher les noms des objets
    --src-prefix <préfixe>
                          afficher le préfixe de source indiqué au lieu de "a/"
    --dst-prefix <préfixe>
                          afficher le préfixe de destination indiqué au lieu de "b/"
    --line-prefix <préfixe>
                          préfixer toutes les lignes en sortie avec la chaîne indiquée
    --no-prefix           n'afficher aucun préfixe, ni de source, ni de destination
    --inter-hunk-context <n>
                          afficher le contexte entre les sections à concurrence du nombre de ligne indiqué
    --output-indicator-new <caractère>
                          spécifier le caractère pour indiquer une nouvelle ligne au lieu de '+'
    --output-indicator-old <caractère>
                          spécifier le caractère pour indiquer une ancienne ligne au lieu de '-'
    --output-indicator-context <caractère>
                          spécifier le caractère pour indiquer une ligne de contexte au lieu de ' '·
Options de renommage de diff
    -B, --break-rewrites[=<n>[/<m>]]
                          casser les modifications d'une réécrire complète en paires de suppression et création
    -M, --find-renames[=<n>]
                          détecter les renommages
    -D, --irreversible-delete
                          supprimer la pré-image pour les suppressions
    -C, --find-copies[=<n>]
                          détecter les copies
    --find-copies-harder  utiliser les fichiers non-modifiés comme sources pour trouver des copies
    --no-renames          désactiver la détection de renommage
    --rename-empty        utiliser des blobs vides comme source de renommage
    --follow              continuer à afficher l'historique d'un fichier au delà des renommages
    -l <n>                empêcher la détection de renommage/copie si le nombre de cibles de renommage/copie excède la limite indiquée·
Options de l'algorithme de diff
    --minimal             produire le diff le plus petit possible
    -w, --ignore-all-space
                          ignorer les espaces lors de la comparaison de ligne
    -b, --ignore-space-change
                          ignorer des modifications du nombre d'espaces
    --ignore-space-at-eol
                          ignorer des modifications d'espace en fin de ligne
    --ignore-cr-at-eol    ignore le retour chariot en fin de ligne
    --ignore-blank-lines  ignorer les modifications dont les lignes sont vides
    -I, --ignore-matching-lines <regex>
                          ignorer les modifications dont les lignes correspondent à <regex>
    --indent-heuristic    heuristique qui déplace les limites de sections de diff pour faciliter la lecture
    --patience            générer un diff en utilisant l'algorithme de différence "patience"
    --histogram           générer un diff en utilisant l'algorithme de différence "histogramme"
    --diff-algorithm <algorithme>
                          choisir un algorithme de différence
    --anchored <texte>    générer un diff en utilisant l'algorithme de différence "ancré"
    --word-diff[=<mode>]  afficher des différences par mot, en utilisant <mode> pour délimiter les mots modifiés
    --word-diff-regex <regex>
                          utiliser <regex> pour décider ce qu'est un mot
    --color-words[=<regex>]
                          équivalent à --word-diff=color --word-diff-regex=<regex>
    --color-moved[=<mode>]
                          les lignes déplacées sont colorées différemment
    --color-moved-ws <mode>
                          comment les espaces sont ignorés dans --color-moved·
Autres options diff
    --relative[=<préfixe>]
                          lancé depuis un sous-répertoire, exclure les modifications en dehors et afficher les chemins relatifs
    -a, --text            traiter les fichiers comme texte
    -R                    échanger les entrées, inverser le diff
    --exit-code           sortir un code d'erreur 1 s'il y avait de différences, 0 sinon
    --quiet               désactiver tous les affichages du programme
    --ext-diff            autoriser l'exécution d'un assistant externe de diff
    --textconv            lancer les filtres externes de conversion en texte lors de la comparaison de fichiers binaires
    --ignore-submodules[=<quand>]
                          ignorer les modifications dans les sous-modules lors de la génération de diff
    --submodule[=<format>]
                          spécifier comment les différences dans les sous-modules sont affichées
    --ita-invisible-in-index
                          masquer les entrées 'git add -N' de l'index
    --ita-visible-in-index
                          traiter les entrées 'git add -N' comme réelles dans l'index
    -S <chaîne>           rechercher les différences qui modifient le nombre d'occurrences de la chaîne spécifiée
    -G <regex>            rechercher les différences qui modifient le nombre d'occurrences de la regex spécifiée
    --pickaxe-all         afficher toutes les modifications dans l'ensemble de modifications avec -S ou -G
    --pickaxe-regex       traiter <chaîne> dans -S comme une expression rationnelle POSIX étendue
    -O <fichier>          contrôler l'ordre dans lequel les fichiers apparaissent dans la sortie
    --rotate-to <chemin>  afficher la modification dans le chemin spécifié en premier
    --skip-to <chemin>    sauter la sortie pour le chemin spécifié
    --find-object <id-objet>
                          rechercher les différences qui modifient le nombre d'occurrences de l'objet indiqué
    --diff-filter [(A|C|D|M|R|T|U|X|B)...[*]]
                          sélectionner les fichiers par types de diff
    --output <fichier>    Sortie vers un fichier spécifié·
]
      at expect (node_modules/expect/build/index.js:135:15)
      at Context.<anonymous> (test/init.spec.js:125:15)
      at processImmediate (internal/timers.js:464:21)


```

It's in french:

```
Rejected to value: [Error: fatal: ni ceci ni aucun de ses répertoires parents n'est un dépôt git : .git
```

But it is translated to:

```
Not a git repository (or any of the parent directories): .git
```

Reply:
It's really strange because:
- I build a Docker image from this branch
- I started the Docker container
- and it works fine inside

So I think it comes from my computer.
How do you work on fork ? Did you use: `npm ci && npm link`, in your fork ?

Reply:
It’s probably related to you git version.
Or language. Setting env LANG helps?

Reply:
Nice catch @mshima so the problem comes from the local LANG :

```
17:53:19 in projects/jhipster/generator-jhipster on  v8-init [$] 
➜ export LANG=en_GB

17:53:25 in projects/jhipster/generator-jhipster on  v8-init [$] 
➜ npm run test test/init.spec.js

> generator-jhipster@7.1.0 pretest
> npm run lint


> generator-jhipster@7.1.0 lint
> npm run eslint && npm run ejs-lint


> generator-jhipster@7.1.0 eslint
> eslint .


> generator-jhipster@7.1.0 ejs-lint
> mocha test/ejslint.js



  EJS Lint
    ✔ EJS templates are valid (585ms)


  1 passing (592ms)


> generator-jhipster@7.1.0 test
> mocha --no-insight --forbid-only --parallel "test/init.spec.js"



  JHipster init generator
    with default options
      ✔ should create expected files (3ms)
    with custom prompt values
      and default options
        ✔ should write custom config to .yo-rc.json
      and skipPrompts option
        ✔ should write default values to .yo-rc.json (1ms)
      and existing config
        ✔ should not write custom prompt values
      and askAnswered option on an exiting project
        ✔ should write custom prompt values (1ms)
    with custom
      skipCommitHook option
        ✔ should create expected files (1ms)
    git feature
      with default option
        ✔ should create .git (1ms)
        ✔ should create 1 commit (9ms)
      with skipGit option
        ✔ should not create .git (1ms)
      regenerating
        ✔ should have 1 commit (8ms)
        ✔ should have uncommited files (11ms)


  11 passing (4s)

```

Reply:
unless I'm wrong, it's related to the new lib simple-git. I wonder if we can force a default lang when using it.

Reply:
Yes.
https://github.com/steveukx/git-js#environment-variables

There is a little warning about language:
https://github.com/steveukx/git-js#pull--diff--merge-summary-responses-dont-recognise-any-files

Reply:
Ok fine!
So using `const git = simpleGit({ baseDir: this.destinationPath() }).env('LANG', 'en_GB');` seems to fix my issue

Do you agree with this change? Can I commit it?

Reply:
Yes, I think plain `en` value should be more appropriated.

Reply:
I can't commit any more, not with my computer right now

Reply:
should be fixed now.